Tests/MongoSwiftSyncTests/SpecTestRunner/SpecTest.swift

```diff
@@ -229,25 +231,36 @@ extension SpecTestFile {
             }
         }
 
+        let topologyType = try setupClient.topologyType()
+
+        var mongosClients: [MongoClient]?
+        switch topologyType {
+        case .sharded, .shardedReplicaSet:
+            var opts = setupClientOptions
+            opts.directConnection = true // connect directly to mongoses
+            mongosClients = try MongoSwiftTestCase.getConnectionStringPerHost()
+                .map { try MongoClient.makeTestClient($0.toString(), options: opts) }
+        default:
+            break
+        }
+
         fileLevelLog("Executing tests from file \(self.name)...")
         for var test in self.tests {
             if let keyword = Self.TestType.skippedTestKeywords.first(where: { test.description.contains($0) }) {
                 print("Skipping test \(test.description) due to matched keyword \"\(keyword)\".")
                 continue
             }
-
-            try self.terminateOpenTransactions(using: setupClient)
+            try self.terminateOpenTransactions(using: setupClient, mongosClients: mongosClients)
             try self.populateData(using: setupClient)
 
             // Due to strange behavior in mongos, a "distinct" command needs to be run against each mongos
             // before the tests run to prevent certain errors from ocurring. (SERVER-39704)
-            if MongoSwiftTestCase.topologyType == .sharded,
+            if [.sharded, .shardedReplicaSet].contains(topologyType),
```

Comment:
Since this will likely be quite common, could we add an `isSharded` (or something similar) method to `TopologyType`?

Reply:
nice idea, done ✅